master/internal/core.go

```diff
@@ -738,6 +740,8 @@ func (m *Master) Run(ctx context.Context) error {
 	// Initialize the HTTP server and listen for incoming requests.
 	m.echo = echo.New()
 	m.echo.Use(middleware.Recover())
+	otelConfig()
+	m.echo.Use(otelecho.Middleware("determined-master"))
```

Comment:
Brilliant. One last thing, let's move this down to the bottom of the middleware (line 788 at present) so it times the whole call, not just the inner call.